packages/amazon-bedrock/src/bedrock-chat-language-model.test.ts

```diff
@@ -994,22 +1114,71 @@ describe('doGenerate', () => {
     };
     stopReason?: string;
     trace?: typeof mockTrace;
+    reasoningContent?:
+      | {
+          reasoningContent: {
+            reasoningText: {
+              text: string;
+              signature?: string;
+            };
+          };
+        }
+      | {
+          reasoningContent: {
+            redactedReasoning: {
+              data: string;
+            };
+          };
+        }
+      | Array<
+          | {
+              reasoningContent: {
+                reasoningText: {
+                  text: string;
+                  signature?: string;
+                };
+              };
+            }
+          | {
+              reasoningContent: {
+                redactedReasoning: {
+                  data: string;
+                };
+              };
+            }
+        >;
   }) {
+    // Create a copy of the content array to avoid modifying the default parameter
+    const contentBlocks = [...content];
+
+    // Add tool calls
+    toolCalls.forEach(tool => {
+      contentBlocks.push({
+        type: 'tool_use',
+        id: tool.id ?? 'tool-use-id',
+        name: tool.name,
+        input: tool.args,
+      });
+    });
+
+    // Add reasoning content
+    if (reasoningContent) {
+      if (Array.isArray(reasoningContent)) {
+        reasoningContent.forEach(rc => {
+          contentBlocks.push(rc as any);
+        });
+      } else {
+        contentBlocks.push(reasoningContent as any);
+      }
+    }
```

Comment:
Ideally order should be kept - is there a way to map the content instead?

Reply:
fdb87b5